apiv6/kentikapi/httputil/httputil.go

```diff
@@ -0,0 +1,46 @@
+package httputil
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+// ClientConfig holds configuration for retrying client.
+type ClientConfig struct {
+	RetryMax     *int
+	RetryWaitMin *time.Duration
+	RetryWaitMax *time.Duration
+}
+
+// NewRetryingClient returns new http.Client with request retry strategy.
+// Exponential backoff algorithm is used to calculate delay between retries.
+// Retry-After header of HTTP 429 response is respected while calculating the retry delay.
+//
+// By default following retry policy is used:
+// - Retry on following HTTP status codes: 429, all 5xx except 501.
```

Comment:
We should retry only on 429, 500, 502, 503 and 504. Retrying on any other 5xx error is waste of time.

Reply:
I'll change the default behaviour